pkg/route/ingress/ingress.go

```diff
@@ -567,14 +548,11 @@ func routeMatchesIngress(
 		return false
 	}
 
-	var secret *corev1.Secret
-	if name, ok := referencesSecret(ingress, rule.Host); ok {
-		secret, _ = secretLister.Secrets(ingress.Namespace).Get(name)
-		if secret == nil {
-			return false
-		}
+	tlsConfig, incomplete := tlsConfigForIngress(ingress, rule, tlsSecretIfValid(ingress, rule, secretLister))
+	if route.Spec.TLS != nil && tlsConfig != nil {
+		tlsConfig.InsecureEdgeTerminationPolicy = route.Spec.TLS.InsecureEdgeTerminationPolicy
```

Comment:
What is the purpose of this assignment?  This prevents the controller resetting the route's `spec.tls.insecureEdgeTerminationPolicy` if the user modifies the generated route, right?  Is that desirable?  None of the other field receive this special treatment.

Reply:
There is a pre-existing test that validates this, `TestController_sync/no-op_-_termination_policy_has_been_changed_by_the_user`

Reply:
Yes that is correct. If someone modified the cert on the route, it would break now. It is however something I intuitively would have expected and required e.G. when using cert manager to update the ingress certs. 

I would hope that no one manually changed a routes cert to something else after it was generated, but I guess Hyums Law applies. Is there a place to call out potentially breaking changes?